common/scylla_configure.py

```diff
@@ -160,7 +158,7 @@ def start_scylla_on_first_boot(self):
         default_start_scylla_on_first_boot = self.CONF_DEFAULTS["start_scylla_on_first_boot"]
         if not self.instance_user_data.get("start_scylla_on_first_boot", default_start_scylla_on_first_boot):
             LOGGER.info("Disabling Scylla start on first boot")
-            self.DISABLE_START_FILE_PATH.touch()
```

Comment:
are you sure calling `sysctl stop` would make is not start on next boot ?

Reply:
I think we should call `systemctl disable --now` instead to make sure it won't start on next boot

Reply:
You are right if we want to permanently disable scylla-server.service.
But `start_scylla_on_first_boot=false` sounds like disabling it only on first boot, because it says "on_first_boot".
Which behavior is expected for this option?

Reply:
The idea of the feature, is that user would be able to configure the cluster manually.
So first boot, it's a must it would be stopped.
If we disable the service, we need to document this change, that user need to also enable it, not just start it.

Otherwise we can still use the hack with the touched file

Reply:
Also check out the unittests, I think they are failing.

Reply:
> Also check out the unittests, I think they are failing.

Fixed the test failure (removed assert self.configurator.DISABLE_START_FILE_PATH.exists() since we don't add the file anymore).

Reply:
> The idea of the feature, is that user would be able to configure the cluster manually. So first boot, it's a must it would be stopped. If we disable the service, we need to document this change, that user need to also enable it, not just start it.

So users expected `scylla_on_first_boot=false` means stop scylla on first boot, then auto-start scylla on next boot.
This is what `systemctl stop scylla-server.service` does.
If we change it to `systemctl disable --now`, we need to document to run `systemctl enable scylla-server.service` before reboot, as you described.

> Otherwise we can still use the hack with the touched file

I thought current touched file hack is enough, but #234 reported that systemd tries to restart scylla-server.service again and again, when " the systemd daemon is reloaded or any other service is manipulated (E.g. systemctl restart rsyslog)".
It is probably because scylla-server.service is failure state, not stopped state, so we need to take care for it.

Alternative approach to fix it is, add `ConditionPathExists=!/etc/scylla/ami_disabled` on scylla-server.service.
But it can stop scylla-server.service but requires to remove the touch file before starting scylla-server, so it requires another documentation.